tasks/Task.cpp

```diff
@@ -26,22 +26,15 @@ Task::~Task()
 
 bool Task::configureHook()
 {
-    if (! TaskBase::configureHook())
-        return false;
-
-    lamps.openURI(_port.get());
+    if (!_io_port.get().empty())
+        lamps.openURI(_io_port.value());
+    else
+        throw std::runtime_error("io_port property not set");
 
-    led_list = _led_list.get();
-    light_level_all  = _light_level_all.get();
+    setDriver(&lamps);
 
-    /* Sets initializing properties of the lamps */
-    for (int i = 0; i < led_list.size(); ++i)
-    {
-    	lamps.setPowerUpLightLevel(led_list[i].power_up_light_level, led_list[i].address);
```

Comment:
If you are going to also set the powerup level everytime, you should update the power_up_light_level member aswell. and probably do that in the setLight_level_all too.